### DISCUSSION TITLE
Best Practices for maintaining and updating Vector DBs with Flowise

### DISCUSSION THREAD

**Discussion by marius-smg** [FlowiseAI/Flowise#3175] [Q&A]
Hey,
i am building a RAG app making use of a vector database. I know how to upsert and query a vector db, but i am unsure how it would implement (automated) maintaining and updating the Database. It is mostly structured Data wich comes from a Database that we maintain inhouse. Sometimes new data is added to the database or certain data is deleted or changed. One Way would be to just delete the whole vector database and upsert the new version of the regular database, but to me this doesnt seem like the best approach. 
Do you know of any best practices how to do this? Does Flowise have Features that help me implementing this? 

Thanks for any help!

**Comment by mdlmarkham** [comment]
I'm very interested in this question as well. Does anyone have experiences to share?

**Comment by amansoni7477030** [comment]
Hi @marius-smg @mdlmarkham, Flowise doesn't have a feature to automatically handle this type of task yet, but they are planning to add it soon. If you need it ASAP, you can easily achieve this with a cron job and Python. I’ve already created a similar solution where my Qdrant database updates daily, with new data being added and old data being removed based on a key. A similar approach can be used for Pinecone and other vector databases.
